engine/src/build/toolchain/mac/BUILD.gn

```diff
@@ -13,6 +13,48 @@ import("//build/toolchain/rbe.gni")
 import("//build/toolchain/ccache.gni")
 import("//build/toolchain/toolchain.gni")
 
+declare_args() {
+  # This controls whether whole module optimization is enabled when building
+  # Swift modules. If enabled, the compiler will compile the module as one
+  # unit, generating just one single object file. Otherwise, it will generate
+  # one object file per .swift file. If unspecified, will default to "true"
+  # for official builds, and "false" for all other builds.
+  swift_whole_module_optimization = -1
+
+  # If true, the intermediate build products of swift module compilation will
+  # be kept after the invocation of the swiftc compiler. Otherwise they will
+  # deleted between each invocation.
+  swift_keep_intermediate_files = false
+
+  # If unspecified, will use the toolchain downloaded via deps.
+  swift_toolchain_path = ""
+}
+
+if (swift_whole_module_optimization == -1) {
+  swift_whole_module_optimization = is_official_build
+}
+
+# When implementing tools using Python scripts, a TOOL_VERSION=N env
+# variable is placed in front of the command. The N should be incremented
+# whenever the script is changed, so that the build system rebuilds all
+# edges that utilize the script. Ideally this should be changed to use
+# proper input-dirty checking, but that could be expensive. Instead, use a
+# script to get the tool scripts' modification time to use as the version.
+# This won't cause a re-generation of GN files when the tool script changes
+# but it will cause edges to be marked as dirty if the ninja files are
+# regenerated. See https://crbug.com/619083 for details. A proper fix
+# would be to have inputs to tools (https://crbug.com/621119).
+get_tool_mtime = rebase_path("//build/toolchain/apple/get_tool_mtime.py")
+tool_versions = exec_script(
```

Comment:
Is there any way to avoid unconditionally executing this python script? This will run on *every* invocation of GN, adding 100s of ms to each `et` invocation.

Reply:
Today we have two invocations (ios/ios_sdk.py, Mac/find_sdk.py). This adds a third.

I think what we can do is merge ios/ios_sdk.py, mac/find_sdk.py into a single script to reduce this from three total invocations to one. I'd prefer to do that in a followup (today or tomorrow) since this patch is already pretty huge and introduces enough build change.

Reply:
sgtm

Reply:
Issue for that work: https://github.com/flutter/flutter/issues/167592